pgmoon/init.moon

```diff
@@ -127,15 +128,26 @@ class Postgres
       @database = opts.database
       @port = opts.port
       @password = opts.password
+      @ssl = opts.ssl
+      @ssl_verify = opts.ssl_verify
+      @ssl_required = opts.ssl_required
+      @key = opts.key
+      @cert = opts.cert
+      @cafile = opts.cafile
 
   connect: =>
     @sock = socket.new!
     ok, err = @sock\connect @host, @port
     return nil, err unless ok
 
     if @sock\getreusedtimes! == 0
+      if @ssl
+        success, err = @send_ssl_message!
+        return nil, err unless success
```

Comment:
We also probably want to disconnect too if the server does not support SSL connections:

```moonscript
elseif @ssl_required
    @disconnect!
    nil, "the server does not support SSL connections"
```

Additionally, we are missing the handling of ErrorMessage responses:

> The frontend should also be prepared to handle an ErrorMessage response to SSLRequest from the server. This would only occur if the server predates the addition of SSL support to PostgreSQL. In this case the connection must be closed, but the frontend might choose to open a fresh connection and proceed without requesting SSL.